packages/salesforcedx-vscode-core/src/commands/templates/apexGenerateClass.ts

```diff
@@ -27,20 +29,20 @@ import {
   APEX_TEST_TEMPLATE
 } from './metadataTypeConstants';
 
-let initialized = false;
-let fileNameGatherer: ParametersGatherer<any>;
-let outputDirGatherer: ParametersGatherer<any>;
-let metadataTypeGatherer: ParametersGatherer<any>;
-let templateGatherer: ParametersGatherer<any>;
+let fileNameGatherer: ParametersGatherer<any> | undefined;
+let outputDirGatherer: ParametersGatherer<any> | undefined;
+let metadataTypeGatherer: ParametersGatherer<any> | undefined;
+let templateGatherer: ParametersGatherer<any> | undefined;
 
 export const getParamGatherers = () => {
-  if (!initialized) {
-    fileNameGatherer = new SelectFileName(APEX_CLASS_NAME_MAX_LENGTH);
-    outputDirGatherer = new SelectOutputDir(APEX_CLASS_DIRECTORY);
-    metadataTypeGatherer = new MetadataTypeGatherer(APEX_CLASS_TYPE);
-    templateGatherer = new ApexTestTemplateGatherer(APEX_TEST_TEMPLATE);
-    initialized = true;
-  }
+  fileNameGatherer =
+    fileNameGatherer ?? new SelectFileName(APEX_CLASS_NAME_MAX_LENGTH);
```

Comment:
```suggestion
  fileNameGatherer ??= new SelectFileName(APEX_CLASS_NAME_MAX_LENGTH);
```

Reply:
Great! I've never used that one before. Thanks.